app/views/users/new.html.erb

```diff
@@ -0,0 +1,49 @@
+<% provide :title, t(".signup") %>
+<h1><%= t ".signup" %></h1>
+<div class="row">
+  <div class="col-12">
+    <%= form_with(model: @user, local: true)  do |form| %>
```

Comment:
form_with, form_for, form_tag khác gì nhau e ?

Reply:
form_tag thì dùng trong các trường hợp không có đối tượng(VD như đăng nhập ..v..v)
form_for thì dùng trong các trường hợp có đối tượng(VD như là, tạo mới, cập nhật..v..v)
2 cú pháp trên thì thường được dùng trong phiên bản rails 5 trở về trước
Ngoài ra form for sử dụng form builder field helper để tạo form, còn form_tag thì không.
còn form_with sử dụng cho cả 2 trường hợp trên, nên thuận tiện và tránh sai cú pháp
Link tham khảo 1: https://viblo.asia/p/phan-biet-form-for-form-tag-va-form-with-trong-rails-Eb85ogw452G
Link tham khảo 2: https://viblo.asia/p/cach-su-dung-form-with-vs-form-for-va-form-tag-trong-rails-Az45bnrL5xY